### PR TITLE
Update to Elasticsearch API to support addition of tags during testing and access policy

### PR DESCRIPTION
**Pull request by jrm16020**
These updates came from testing with Terraform.  Terraform would fail to create the Elasticsearch domain because it never saw it as completed.  Making these changes allowed creating an Elasticsearch domain and applying an access policy to that domain for testing.

### CODE REVIEW

File: localstack/services/es/es_api.py [hunk 1]
```diff
@@ -23,6 +23,144 @@ def error_response(error_type, code=400, message='Unknown error.'):
     return response, code
 
 
+def get_domain_config(domain_name):
+    return {
+        'DomainConfig': {
```
Comment (whummer) [inline localstack/services/es/es_api.py]: Wondering if this is the most "minimal" hardcoded domain config that we can return. Some of the configurations appear to be fairly specific to your scenario (e.g., `AdvancedOptions`), so it is unclear how well they generalize to arbitrary use cases.

Also, there are 9 `'Status'` dicts that are very similar in this configuration - would it make sense to have a small parameterizable utility function that returns a corresponding `Status` dict?

**Comment by whummer** [comment]
Thanks for this PR @jrm16020 ! Looks like there is currently a merge conflict - can we please rebase this branch onto the latest `master`. Thanks

**Comment by jrm16020** [comment]
Thank you for the feedback @whummer .  I'll rebase and address the comments soon.

File: localstack/services/es/es_api.py [hunk 2]
```diff
@@ -34,14 +172,20 @@ def get_domain_status(domain_name, deleted=False):
             'ElasticsearchClusterConfig': {
                 'DedicatedMasterCount': 1,
                 'DedicatedMasterEnabled': True,
-                'DedicatedMasterType': 'm3.medium.elasticsearch',
+                'DedicatedMasterType': 'm4.large.elasticsearch',
```
[the remark this answers is not in the document]
Comment (jrm16020) [reply localstack/services/es/es_api.py]: There's no good reason.  I made everything match what I'd used for the `get_domain_config` function.  I'll revert back to m3.medium for now to keep this PR focused on the functions I needed to replicate for Terraform to work.

File: localstack/services/es/es_api.py (hunk 1 above)
Comment (jrm16020) [reply localstack/services/es/es_api.py]: I copied the JSON from what's documented here:
https://docs.aws.amazon.com/elasticsearch-service/latest/developerguide/es-configuration-api.html#es-configuration-api-actions-describeelasticsearchdomainconfig

I can get rid of some of the more specific options if it helps.

I'll get rid of the duplicated `'Status'` dicts with a small function as well.

File: tests/integration/test_elasticsearch.py [hunk 3]
```diff
@@ -66,7 +67,14 @@ def test_domain_creation():
     status = es_client.describe_elasticsearch_domain(DomainName=TEST_DOMAIN_NAME)
     assert_equal(status['DomainStatus']['DomainName'], TEST_DOMAIN_NAME)
     assert_true(status['DomainStatus']['Created'])
+    assert_false(status['DomainStatus']['Processing'])
     assert_false(status['DomainStatus']['Deleted'])
+    assert_equal(status['DomainStatus']['Endpoint'], TEST_ENDPOINT_URL)
```
Comment (whummer) [inline tests/integration/test_elasticsearch.py]: The tests in Travis-CI are currently failing at this line. Please also use `aws_stack.get_elasticsearch_endpoint()` here and remove the `TEST_ENDPOINT_URL` variable.

**Comment by whummer** [comment]
We're almost there @jrm16020 - only one minor issue remaining in the tests (see comment above). Thanks

**Comment by jrm16020** [comment]
Sorry for the delay @whummer .  Work has been **very** busy!  I made the change you were asking about.  Do you want me to squash commits or do anything else?

**Comment by whummer** [comment]
Great, thanks for the quick turnaround. Squashed and merged this PR 👍

**Pull request merged**